src/memex/search/index.py

```diff
@@ -92,25 +119,16 @@ class BatchIndexer(object):
     the search index.
     """
 
-    def __init__(self, session, es_client, request):
+    def __init__(self, session, es_client, request, target_index=None):
         self.session = session
         self.es_client = es_client
         self.request = request
 
         # By default, index into the open index
-        self._target_index = self.es_client.index
-
-    def index_all(self):
```

Comment:
Agreed, but the actual functionality of `BatchIndexer` is currently split between a) code to return a stream of annotations as fast as possible from a query, and b) code to manage indexing (including, possibly, retries). I'll try and clean this up in subsequent PRs.